helix-term/src/args.rs

```diff
@@ -41,15 +41,43 @@ impl Args {
                         }
                     }
                 }
-                arg => args.files.push(PathBuf::from(arg)),
+                arg => args.files.push(parse_file(arg)),
             }
         }
 
         // push the remaining args, if any to the files
-        for filename in iter {
-            args.files.push(PathBuf::from(filename));
+        for arg in iter {
+            args.files.push(parse_file(arg));
         }
 
         Ok(args)
     }
 }
+
+/// Parse arg into [`PathBuf`] and position.
+pub(crate) fn parse_file(s: &str) -> (PathBuf, usize, usize) {
+    split_path_row_col(s)
+        .or_else(|| split_path_row(s))
+        .unwrap_or_else(|| (PathBuf::from(s), 0, 0))
+}
+
+/// Split file.rs:10:2 into [`PathBuf`], row and col.
+///
+/// Does not validate if file.rs is a file or directory.
+fn split_path_row_col(s: &str) -> Option<(PathBuf, usize, usize)> {
```

Comment:
This should use `Option<(PathBuf, Position)>`: https://github.com/helix-editor/helix/blob/cffd0f8a6a02cf153159be608a7d1b3669a92bc4/helix-core/src/position.rs

Reply:
Thanks, I forgot about that.